Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp

```diff
@@ -0,0 +1,76 @@
+#include <WarpXAlgorithmSelection.H>
+#include<FiniteDifferenceAlgorithms/YeeAlgorithm.H>
+#include<FiniteDifferenceSolver.H>
+#include<AMReX_Gpu.H>
+
+using namespace amrex;
+
+void FiniteDifferenceSolver::EvolveB ( VectorField& Bfield,
+                                       VectorField& Efield,
+                                       amrex::Real dt ) {
+    // Select algorithm (The choice of algorithm is a runtime option,
+    // but we compile code for each algorithm, using templates)
+    if (m_fdtd_algo == MaxwellSolverAlgo::Yee){
+        EvolveBwithAlgo <YeeAlgorithm> ( Bfield, Efield, dt );
+//    } else if (fdtd_algo == MaxwellSolverAlgo::CKC) {
+//       EvolveBwithAlgo <CKCAlgorithm> ( Bfield, Efield, dt );
+    } else {
+        amrex::Abort("Unknown algorithm");
+    }
+}
+
+template<typename algo>
```

Comment:
It's usually good practice, since they are types, to write template parameters with an upper case first letter.
e.g. `Algo` or even better `T_Algo`. Ideally even `T_FieldSolver` for good readability.